pkg/services/folder/folderimpl/folder.go

```diff
@@ -1004,14 +1008,36 @@ func (s *Service) deleteChildrenInFolder(ctx context.Context, orgID int64, folde
 }
 
 func (s *Service) legacyDelete(ctx context.Context, cmd *folder.DeleteFolderCommand, folderUIDs []string) error {
+	// if dashboard restore is on we don't delete public dashboards, the hard delete will take care of it later
+	if !s.features.IsEnabledGlobally(featuremgmt.FlagDashboardRestore) {
```

Comment:
You'll also need to add this to `deleteFromApiServer` in `folder_unifiedstorage.go` for this to work in k8s